### PR TITLE
Changes to make the lib compatible with godot 4.0+

### PR DESCRIPTION
**Pull request by olixis**
# Update for Godot 4.0+ Compatibility

## Description

This update modifies the `godot-polygon2d-fracture` library to ensure compatibility with Godot 4.0+. The key changes include:

- Adjustments to scripts and scenes to align with the new Godot 4.0+ API.
- Updated `.gitignore` to include Godot 4.0+ specific files and directories.

### .gitignore

# Godot-specific ignores
.import/
.godot/
export.cfg
export_presets.cfg
# Mono-specific ignores
.mono/
data_/
### project.godot
config/features=PackedStringArray("4.2")
# README.md
UPDATE v2.0.0 ported to Godot 4.0
# Testing
- Tested with Godot 4.2.2 to ensure compatibility and functionality.
- Verified that all demo scenes and scripts work as expected.
# Notes
- The library now requires Godot 4.0+ to function correctly.
- Feedback and further testing are appreciated to ensure full compatibility.

Thank you for considering this update!

### CODE REVIEW

File: demo/src/Fracture.gd [hunk 1]
```diff
@@ -181,5 +181,5 @@ func _on_FracturesSlider_value_changed(value: float) -> void:
 
 
 func _on_MinAreaSlider_value_changed(value):
-	_min_area_label.text = "Min Area: %d" % value
```
Comment (SoloByte) [inline demo/src/Fracture.gd]: I think this should still be Area instead of Area3D?

**Comment by SoloByte** [comment]
Thanks a lot for doing this :) 
I have not done anything in Godot for a while and I have not used Godot 4 at all, so thanks again for the work. I will take a look and test it as soon as possible.

**Comment by SoloByte** [comment]
I have made a branch for 1.4.x developement in case someone wants to use it / improve it for Godot 3.

File: demo/src/Fracture.gd (hunk 1 above)
Comment (olixis) [reply demo/src/Fracture.gd]: Sure! I'm don't really know why it was changed.

File: demo/src/Main.gd [hunk 2]
```diff
@@ -62,14 +62,14 @@ func changeTest() -> void:
...
+	var instance = test_scenes[_cur_test_scene_index].instantiate()
 	add_child(instance)
 	
 	#for some reason I need to do that in Godot v3.3.1...
 	#the camera in the scene is already setup with current = true and zoom = Vector2(2,2)
 	#but it does not work without setting it here again
-	instance.get_node("Camera2D").current = true
-	instance.get_node("Camera2D").zoom = Vector2(2,2)
+	instance.get_node("Camera2D").make_current()
+	instance.get_node("Camera2D").zoom = Vector2(0.25,0.25)
```
Comment (SoloByte) [inline demo/src/Main.gd]: camera zoom should be (0.5, 0.5) to work properly? At least for me 0.25 is too much zoomed out.

Screenshot 2024-06-21 at 08 35 52

**Comment by SoloByte** [comment]
The only other issues I found:

- [x] The UI seems to be placed/ anchored wrong
- [x] CutFracture Scene (1 scene when started) seems to have problems with the textures on the polygons

I am currently investigating it but I am not up to date with Godot anymore...

**Comment by SoloByte** [comment]
If you want to I can commit the fixes for the UI, the camera, and the texture problem to the PR. The commit would resolve all issues I found and I would merge the pull request after that.

File: demo/src/Main.gd (hunk 2 above)
Comment (SoloByte) [reply demo/src/Main.gd]: See comments below.

**Comment by olixis** [comment]
> If you want to I can commit the fixes for the UI, the camera, and the texture problem to the PR. The commit would resolve all issues I found and I would merge the pull request after that.

For some reason, i'm not having the listed issues on my machine. But sure! Please, commit the fixes.

**Comment by SoloByte** [comment]
That's why I asked, because I tested it on windows & macOS and I had those issues both times...

**Comment by SoloByte** [comment]
You could download your version as a zip file and test it as a new project in Godot? 

I am mainly interested in the camera zoom issue and the UI issues because I don't want to commit fixes for a problem that may not exist ;)

**Comment by etherealxx** [comment]
Damn i should've read this first. I thought this addon was abandoned.

I also made one😭
[godot4-polygon2d-fracture.zip](https://github.com/user-attachments/files/15981766/godot4-polygon2d-fracture.zip)

**Comment by SoloByte** [comment]
> Damn i should've read this first. I thought this addon was abandoned.
> 
> I also made one😭 [godot4-polygon2d-fracture.zip](https://github.com/user-attachments/files/15981766/godot4-polygon2d-fracture.zip)

I guess there will be more ^^ It's not abandoned. I am waiting for Godot to improve its custom drawing performance before I am using it again. Until then I am just doing other stuff ;)

**Review by SoloByte — APPROVED** [review]

**Comment by etherealxx** [comment]
i think you should put a notice on the readme for godot 3 user to use the other branch?
I mean like more clearer notice

**Comment by SoloByte** [comment]
> i think you should put a notice on the readme for godot 3 user to use the other branch? I mean like more clearer notice

I will do that!

**Pull request merged**